### DISCUSSION TITLE
Option for control channel data packets during an ongoing test

### DISCUSSION THREAD

**Discussion by d3cxxxx** [esnet/iperf#1675] [Ideas]
Our iperf3 server is behind a network appliance that shuts off any client that is idle for a fixed number of seconds. 
When we start a long running UDP test, the iperf3 client establishes a control channel, performs the initial exchange and stays quiet during the test. Exactly when the test reached the timeout seconds (as configured in the network appliance), it closes the client (even though the UDP test is running without issues) that causes the test to break. 
The network appliance does not allow the use of tcp keepalive packets.

So, the request is if it makes sense and is possible to allow an iperf3 configuration that sends a 1-byte packet over the control channel at a configurable interval that can be sent (either by client or server) and discarded by the other entity. This is for cases where a TCP keepalive option does not work.

Apologies if such an option already exists and I missed it.

**Comment by davidBar-On** [comment]
I think this is a good idea.  As far as I know there is no such option/solution for iperf3.

PR #1423 solves the issue by adding support for TCP Keepalive.  However, it may be that TCP keepalive mechanism is not supported by all TCP stacks, so the idea suggested here may be better as it will work for all cases.  I assume the added overhead is negligible.
